pandas/core/internals.py

```diff
@@ -3696,6 +3720,11 @@ def is_datelike_mixed_type(self):
         self._consolidate_inplace()
         return any(block.is_datelike for block in self.blocks)
 
+    @property
+    def any_extension_types(self):
```

Comment:
possible nitpick, but is `any_extension_type` (without s) grammatically more correct?

Reply:
hmm, we should not actually care about this, what is the reason?

Reply:
See the change to `NDFrame.count`. Without it there's an edge case for single-block DataFrames, since we do

```python
            if frame._is_mixed_type or frame._data.any_extension_types:
                result = notna(frame).sum(axis=axis)
            else:
                counts = notna(frame.values).sum(axis=axis)  # This .values is bad for EA
                result = Series(counts, index=frame._get_agg_axis(axis))
```

Taking the `self.values` coerces to objects, which will have incorrect `isna` behavior.

Reply:
@jorisvandenbossche I with with `types` since you could have multiple kinds of extension array types in a DataFrame :)